src/graphql-types/ReasonApolloQuery.re

```diff
@@ -8,6 +8,23 @@ module Get = (Config: ReasonApolloTypes.Config) => {
     | Loading
     | Error(apolloError)
     | Data(Config.t);
+  type updateQueryOptions = {
```

Comment:
If you change this to
```ocaml
[@bs.deriving abstract]
  type updateQueryOptions = {
    fetchMoreResult: option(Js.Json.t),
    variables: Js.Json.t,
  };
```
you can remove the next type declaration as they are then the same.

And you can simplify the switch at line 122 to:
```ocaml
            switch (updateQuery) {
            | Some(updateQuery) =>
              Some(
                (
                  (previousResult, opts) => updateQuery(previousResult, opts)
                ),
              )

            | None => None
            }
```

Reply:
That seems to have a small side effect. In the application code that is using this: opts##fetchMoreResult is no longer optional. It seems like it should be based on: https://www.apollographql.com/docs/react/features/pagination.html#numbered-pages shouldn't it?